### PR TITLE
Added Cell counting Tutorial

### PR DESCRIPTION
**Pull request by aaronrockmenezes**
## Description

Fix #(issue)

Added Cell Counting Tutorial
- [x] Need to add more docs
- [x] Need to add citation for tutorial
- [x] Add link to Colab
- [x] Improve comments
- [x] Add explanation
- [x] Cite Molnet and BBBC
- [x] Linting fixes

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: examples/tutorials/Cell_Counting_Tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,503 @@
+{
```
Comment (rbharath) [inline examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: <p>Start with an introduction paragraph. Intro the problem, and why the reader should care</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/3845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (rbharath) [inline examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: <p>Introduce the dataset. Make a new section. What is this dataset? Add a citation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/3845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (rbharath) [inline examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: <p>Add a diagram for the CNN. More detail about the model. Make this a new section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/3845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (rbharath) [inline examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: <p>Link to another tutorial that introduces callbacks</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/3845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (aaronrockmenezes) [reply examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: Done. Please review

Comment (aaronrockmenezes) [reply examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: Done. Please review

Comment (aaronrockmenezes) [reply examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: Done. Please review

File: examples/tutorials/Cell_Counting_Tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
```
Comment (rbharath) [inline examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: <p>Small typo "\"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/3845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (aaronrockmenezes) [reply examples/tutorials/Cell_Counting_Tutorial.ipynb:1]: Done.

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**